ehrql/backends/tpp.py

```diff
@@ -89,15 +89,21 @@ class TPPBackend(BaseBackend):
         """
             SELECT
                 Patient_ID AS patient_id,
-                CAST(ConsultationDate AS date) AS date,
+                CASE
+                    WHEN ConsultationDate = '9999-12-31T00:00:00' THEN NULL
+                    ELSE CAST(ConsultationDate AS date)
+                END AS date,
```

Comment:
Since the events of the last few weeks, I'm now super nervous about changes to the query structure of large, widely-used tables like this. I'm not saying this one in particular is likely to cause problems. But I'd like us to do at least some basic performance testing on this query to make sure it performs OK when filtering by date.

There are also a few equivalent ways of writing this which it would be good to compare the performance of e.g.
```
CAST(NULLIF(ConsultationDate, '9999-12-31T00:00:00')) AS date
```

Doing this is painful in the sense that it requires manually constructing SQL queries in the console and using `SET STATISTICS TIME ON` to compare CPU times (the only metrics we have access to). But I don't see an alternative at the moment.

Reply:
@evansd Based on investigation in [#1523](https://github.com/opensafely-core/ehrql/issues/1523#issuecomment-1706844168) it looks like NULLIF may be marginally more efficienct, so I've updated it to use that. And also as discussed there, it doesn't seem like this change is going to make things worse